Gems/MotionMatching/Code/Source/FeatureAngularVelocity.cpp

```diff
@@ -0,0 +1,158 @@
+    {
+        const AZ::Vector3 queryVelocity = context.m_queryVector.GetVector3(m_featureColumnOffset);
+        const AZ::Vector3 frameVelocity = context.m_featureMatrix.GetVector3(frameIndex, m_featureColumnOffset);
+
+        return CalcResidual(queryVelocity, frameVelocity);
+    }
+
+    void FeatureAngularVelocity::DebugDraw(
+        AzFramework::DebugDisplayRequests& debugDisplay,
+        const Pose& pose,
+        const AZ::Vector3& velocity,
+        size_t jointIndex,
+        size_t relativeToJointIndex,
+        const AZ::Color& color)
+    {
+        const Transform jointModelTM = pose.GetModelSpaceTransform(jointIndex);
+        const Transform relativeToWorldTM = pose.GetWorldSpaceTransform(relativeToJointIndex);
+
+        const AZ::Vector3 jointPosition = relativeToWorldTM.TransformPoint(jointModelTM.m_position);
+        const AZ::Vector3 velocityWorldSpace = relativeToWorldTM.TransformVector(velocity);
+
+        DebugDrawVelocity(debugDisplay, jointPosition, velocityWorldSpace * mm_debugDrawVelocityScale, color);
+    }
+
+    void FeatureAngularVelocity::DebugDraw(
+        AzFramework::DebugDisplayRequests& debugDisplay,
+        const Pose& currentPose,
+        const FeatureMatrix& featureMatrix,
+        const FeatureMatrixTransformer* featureTransformer,
+        size_t frameIndex)
+    {
+        if (m_jointIndex == InvalidIndex)
+        {
+            return;
+        }
+
+        AZ::Vector3 angularVelocity = featureMatrix.GetVector3(frameIndex, m_featureColumnOffset);
+        if (featureTransformer)
+        {
+            angularVelocity = featureTransformer->InverseTransform(angularVelocity, m_featureColumnOffset);
+        }
+        DebugDraw(debugDisplay, currentPose, angularVelocity, m_jointIndex, m_relativeToNodeIndex, m_debugColor);
+    }
+
+    void FeatureAngularVelocity::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (!serializeContext)
+        {
+            return;
+        }
+
+        serializeContext->Class<FeatureAngularVelocity, Feature>()->Version(1);
+
+        AZ::EditContext* editContext = serializeContext->GetEditContext();
+        if (!editContext)
+        {
+            return;
+        }
+
+        editContext->Class<FeatureAngularVelocity>("FeatureAngularVelocity", "Matches joint angular velocities.")
+            ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+            ->Attribute(AZ::Edit::Attributes::AutoExpand, "");
+    }
+
+    size_t FeatureAngularVelocity::GetNumDimensions() const
+    {
+        return 3;
+    }
+
+    AZStd::string FeatureAngularVelocity::GetDimensionName(size_t index) const
+    {
+        AZStd::string result = m_jointName;
+        result += '.';
+
+        switch (index)
+        {
+        case 0:
+            {
+                result += "AngularVelocityX";
```

Comment:
nit: Potentially using a `AZStd::string::format` here might be more optimal but might not be worth the refactor depending on how often this is called

Reply:
I was thinking you could store it in an array and use the index to append to your string. 

```
arr = ["AngularVelocityX", "AngularVelocityY", "AngularVelocityZ"];
for(int i = 0; i < len(arr) && i <= index; ++i) {
 result += arr[i];
}
```